pkg/metrics/observers/kuma.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2021 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package observers
+
+import (
+	"fmt"
+	"time"
+
+	flaggerv1 "github.com/fluxcd/flagger/pkg/apis/flagger/v1beta1"
+	"github.com/fluxcd/flagger/pkg/metrics/providers"
+)
+
+// TODO [@johnharris85]: Do we also need to select by mesh here? These could be duplicated (but in different meshes).
```

Comment:
I don't think duplication is possible since the metric refers to a service by name and namespace which are unique per cluster. Can you add the same service to two Kuma meshes?